### DISCUSSION TITLE
Are there any plans to support the Transitland V2 API?

### DISCUSSION THREAD

**Discussion by newcarrotgames** [valhalla/valhalla#4009] [Q&A] [answered]
I'm attempting to build the tiles using the Transitland API. Every time I run the Transitland import script it just says 0 of 24 feeds imported (or something similar, that's from my feeble memory). I noticed that the response is empty when it tries to request data from Transitland. I researched and found that the V1 API is no longer supported and everyone should start using the V2 API. The V2 API does not support the bounding box parameter, but the data should be the same just in a different format, so I was wondering if anyone has tried to adapt the V1 Transitland script to the V2 format.

**Comment by kevinkreiser** [comment, marked as answer]
back in the day there was a plan for transitland to publish pbf tiles (the same that our internal processing makes out of the v1 api). if the v2 api ended up being these pbf tiles, then we will still support that. in fact, we are currently in the middle of adding support for GTFS directly and that involves creating these pbf tiles as an intermediate step (we load GTFS into these pbf tiles).

so yeah long story short, if its pbf tiles we can support it. if its something else and someone has the time to refactor `valhalla_fetch_transit` to create these pbf tiles out of calls to the v2 api, we would also be happy to support that as well. in the mean time though, the maintainers dont have any plans to migrate the transit fetcher to transitland v2 ourselves.

**Comment by nilsnolde** [comment]
It doesn't seem like v2 returns PBFs: https://www.transit.land/documentation/rest-api/stops#response-format. It's still a JSON response which will need to be parsed very similar to v1 (at least for Valhalla), possibly just another data model?

So yeah, I guess that won't be supported anytime soon, unless someone is willing to port the v1 logic (before we added native GTFS support) to v2 and of course maintain it.

Also, eventually I'd personally like to get away from the intermediate PBF generation, as the step writing those (i.e. `valhalla_ingest_traffic`) is the bottleneck right now, it's really slow.. I'm not sure why it was done like this in the first place, but as of now it doesn't seem very necessary. I realize it's easier to work with than the graph builders, but that's a small pain compared to the penalty of writing the same data once more in PBF.
